client/components/notes-formatted-block/index.jsx

```diff
@@ -35,7 +36,10 @@ export const FormattedBlock = ( { content = {} } ) => {
 	switch ( type ) {
 		case 'a':
 		case 'link':
-			return <a href={ content.url }>{ descent }</a>;
+			const url = startsWith( content.url, 'https://wordpress.com/' )
+				? content.url.substr( 21 )
+				: content.url;
+			return <a href={ url }>{ descent }</a>;
```

Comment:
Nice work addressing this! 
For more context on the reasoning for this, the activity log endpoint returns canonical URLs for links, which can break the experience when you are in non-production environments.
This bit of code will fix it so that AL links will respect the current env.